libs/designsystem/src/lib/components/chart/chart-js/chart-js.service.ts

```diff
@@ -163,36 +203,95 @@ export class ChartJSService {
   }
 
   private createOptionsObject(args: {
-    type: ChartType;
     customOptions?: ChartOptions;
     annotations?: AnnotationOptions[];
+    chartPeriod?: ChartPeriod;
+    dataLabelOptions?: ChartDataLabelOptions;
   }): ChartOptions {
-    const { type, customOptions, annotations } = args;
+    const {
+      customOptions,
+      annotations,
+      chartPeriod,
+      dataLabelOptions: chartDataLabelOptions,
+    } = args;
 
-    const typeConfig = this.chartConfigService.getTypeConfig(type);
+    const typeConfig = this.chartConfigService.getTypeConfig(this.chartType);
     const typeConfigOptions = typeConfig?.options;
     const annotationPluginOptions = annotations
       ? this.createAnnotationPluginOptionsObject(annotations)
       : {};
+
+    let defaultOptions: ChartOptions = {
+      locale: this.locale,
+      plugins: { tooltip: { callbacks: {} } },
+      scales: { y: { ticks: {} } },
+    };
+
+    if (this.chartType === 'stock') {
+      defaultOptions.plugins.tooltip.callbacks.label = (context) => {
+        // It's not possible to add spacing between color legend and text so we
+        // prefix with a space.
+        return ' ' + context.formattedValue + (chartDataLabelOptions.valueSuffix || '');
+      };
+
+      defaultOptions.scales.y.ticks.callback = (value) => {
+        return value + (chartDataLabelOptions.valueSuffix || '');
+      };
+
+      defaultOptions.plugins.tooltip.callbacks.title = (tooltipItems) => {
+        const date = toDate((tooltipItems[0]?.raw as any)?.x);
+        if (date.valueOf()) {
+          return date.toLocaleTimeString(this.locale, {
+            day: 'numeric',
+            month: 'short',
+            hour: '2-digit',
+            minute: '2-digit',
+          });
+        }
+      };
```

Comment:
Could it make sense to package these into a `createStockChartOptions` method that returns a new ChartOption object that then is merged with the others, as is done with annotationPluginOptions on line 220?